R/likelihood.R

```diff
@@ -78,9 +128,36 @@ likelihood <- function(chains, statistic = c("size", "length"), offspring_dist,
   checkmate::assert_numeric(
     exclude, null.ok = TRUE
   )
+  # likelihood cannot work with an <epichains> object so convert to
+  # <epichains_summary>
+  if (.is_epichains(chains)) {
+    chains <- summary(chains)
+  }
+
+  # If stat_threshold is not specified, try to get it from the chains object
+  # or set it to Inf if chains is numeric
+  if (missing(stat_threshold)) {
+    if (.is_epichains_summary(chains)) {
+      stat_threshold <- attr(chains, "stat_threshold")
+    } else if (is.numeric(chains)) {
+      stat_threshold <- Inf
+    }
+  }
+ # If stat_threshold is specified, it cannot be NULL
+  if (is.null(stat_threshold)) {
+    stop("`stat_threshold` must be an integer.")
+  }
+ # Replace infinite chain sizes with stat_threshold
+  chains[is.infinite(chains)] <- stat_threshold
```

Comment:
Doesn't this cause issues before https://github.com/epiverse-trace/epichains/issues/243 is addressed, i.e. we could have chains longer than `stat_threshold`?
```suggestion
if (is.finite(stat_thrsehold)) {
  chains <- pmin(chains, stat_threshold)
}
```

Reply:
Thanks.